### PR TITLE
PowerShell based installation script for Windows nodes.

### PR DESCRIPTION
**Pull request by phillipsj**
This provides a PowerShell based installation script for Windows nodes.

Signed-off-by: Jamie Phillips <jamie.phillips@suse.com>

#### Proposed Changes ####

Adds an install.ps1 script to support installation on Windows nodes.

#### Types of Changes ####

A new feature that will not cause any breaking changes.

#### Linked Issues ####

* Issue #1240
* https://github.com/rancher/windows/issues/55
* https://github.com/rancher/windows/issues/50
* https://github.com/rancher/windows/issues/51

### CODE REVIEW

File: install.ps1 [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+.EXAMPLE
+  Usage:
+    Invoke-WebRequest ((New-Object System.Net.WebClient).DownloadString('https://get.rke2.io/install.ps1'))        
+    ./install.ps1 -Channel Latest -Mehtod Tar
+#>
+
+    [CmdletBinding()]
+param (
+    [Parameter()]
+    [ValidateSet("latest", "stable", "testing", "v1.20", "v1.19", "v1.18")]
```
Comment (brandond) [inline install.ps1:49]: This should come from the channel server metadata, but if you're going to hardcode it...

```suggestion
    [ValidateSet("latest", "stable", "testing", "v1.21", "v1.20", "v1.19"]
```

File: install.ps1 [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+        Write-FatalLog "You need to be administrator to perform this install"
+    }
+
+    Write-Host "Using $($Channel) channel of rke2 for installation"
+
+    # --- install tarball to /usr/local by default, except if /usr/local is on a separate partition or is read-only
+    # --- in which case we go into /opt/rke2.
+    if (Test-TargetMountpoint -and Test-TargetRo) {
+        $TarPrefix="C:\opt\rke2"
+        Write-Warn $DefaultTarPrefix + "is read-only or a mount point; installing to " + $TarPrefix
```
Comment (brandond) [inline install.ps1:144]: Is this even a thing on Windows?

File: install.ps1 [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+        [Parameter()]
+        [string]
+        $InstallPath,
+        [Parameter()]
+        [String]
+        $Tarball
+    )
+    Write-InfoLog "unpacking tarball file to $InstallPath"
+    New-Item -Path $InstallPath -Type Directory -Force
+    tar xzf "$Tarball" -C "$InstallPath"
```
Comment (brandond) [inline install.ps1:369]: No cmdlet for extracting tarballs, huh

File: install.ps1 [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+    )    
+
+    if ($ExpectedAirGapChecksum) {
+        return
+    }
+    New-Item -Path "$InstallAgentImageDir" -ItemType "Directory"
+    $archInfo = Get-ArchitectureInfo
+    $suffix = $archInfo.Suffix
+
+    # releases that provide zst artifacts can read from the compressed archive; older releases
```
Comment (brandond) [inline install.ps1:534]: The old non-zst-supporting releases won't have Windows builds, so you don't need to worry about this. Just use zst for everything.

**Review by brandond — CHANGES_REQUESTED** [review]

File: install.ps1 (hunk 3 above)
Comment (phillipsj) [reply install.ps1:369]: No there isn't a commandlet for extracting tarballs.

File: install.ps1 (hunk 4 above)
Comment (phillipsj) [reply install.ps1]: Good call, removed.

File: install.ps1 (hunk 2 above)
Comment (phillipsj) [reply install.ps1]: We just ported applying the same logic. I think that is a good point and it was removed.

File: install.ps1 [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+    If (-NOT $currentRole.IsInRole($adminRole))
+    {
+        Write-FatalLog "You need to be administrator to perform this install"
+    }
+
+    Write-Host "Using $($Channel) channel of rke2 for installation"
+}
+
+# check_method_conflict will exit with an error if the user attempts to install
+# via tar method on a host with existing RPMs.
```
Comment (brandond) [inline install.ps1:119]:
```suggestion
# via tar method on a host with existing chocolatey package.
```

**Review by brandond — APPROVED** [review]
LGTM, although I can't test myself as I don't have a windows box.

One comment nit.

**Review by rosskirkpat — APPROVED** [review]
LGTM

**Review by luthermonson — APPROVED** [review]

**Pull request merged**